fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -488,6 +488,9 @@ protected OptExpression rewriteProjection(RewriteContext rewriteContext,
         for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : swappedQueryColumnMap.entrySet()) {
             ScalarOperator rewritten = replaceExprWithTarget(entry.getValue(),
                     normalizedViewMap, rewriteContext.getOutputMapping());
+            if (rewritten == null) {
```

Comment:
What's the situation of being null?

Reply:
if the entry.getValue can not be rewritten by normalizedViewMap's keys, it will return null.
for example. mv has output columns a, b. query has output columns c. then it will return null.